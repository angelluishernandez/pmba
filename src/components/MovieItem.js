import React from "react";
import { ButtonContainer } from "./styled-components/ButtonContainer";
import { MovieItemWrapper } from "./styled-components/MovieItemWrapper";
import { setToWatched } from "../redux/actions/movies.actions";
import { connect } from "react-redux";
import { FormItemWrapper } from "./styled-components/FormItemWrapper";

const MovieItem = ({ movie, setToWatched }) => {
	const handleRadioChange = (e) => {
		setToWatched(movie.id);
	};

	return (
		<>
			{movie === undefined && movie !== [] ? (
				<div>Loading...</div>
			) : (
				<MovieItemWrapper>
					<div>
						<h3>{movie.name}</h3>
					</div>
					<div>
						<h3>Genres</h3>
						<ul>
							{movie.movieGenres.map((genre, index) => (
								<li>{genre}</li>
							))}
						</ul>
					</div>

					<FormItemWrapper>
						<div>
							<label>Have you watched it already?</label>
						</div>
						<input
							type="checkbox"
							value="hasBeenWatched"
							checked={movie.watched}
							onChange={(e) => handleRadioChange(e)}
						/>
						<ButtonContainer delete>Delete</ButtonContainer>
					</FormItemWrapper>
				</MovieItemWrapper>
			)}
		</>
	);
};

const mapDispatchToProps = (dispatch) => ({
	setToWatched: (movieName) => dispatch(setToWatched(movieName)),
});

export default connect(undefined, mapDispatchToProps)(MovieItem);
