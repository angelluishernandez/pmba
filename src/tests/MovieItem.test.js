import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import React from "react";
import configureMockStore from "redux-mock-store";

import MovieItem from "../components/MovieItem";

configure({ adapter: new Adapter() });
const middlewares = [];
const createMockStore = configureMockStore(middlewares);

test("should render MovieItem correctly", () => {
	const initialState = { movies: [], filters: [] };
	const store = createMockStore(initialState);
	const wrapper = shallow(<MovieItem store={store} />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
