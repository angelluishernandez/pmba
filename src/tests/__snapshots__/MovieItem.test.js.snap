// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieItem correctly 1`] = `
<MovieItem
  setToWatched={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
